docusaurus/docs/Android/04-compose/07-guides/02-customizing-image-and-video-previews.mdx

```diff
@@ -0,0 +1,166 @@
+# Customizing Image and Video Previews
+
+## Introduction
+
+The Compose UI Components chat SDK supports various types of attachments by using [attachment factories](../03-general-customization/01-chat-theme.mdx#streamattachmentfactories). The one used to display image and video attachments is called `MediaAttachmentFactory` and offers a degree of customization.
+
+:::note
+Video Attachment preview thumbnails are a paid feature and as such can be turned off. You can find the pricing [here](https://getstream.io/chat/pricing/) and the information on how to turn them off in the [this](../05-message-components/03-message-list.mdx#image-and-video) section.
+:::
+
+## Customization
+
+Let's take a look at the function designed to produce the necessary attachment factory:
+
+```kotlin
+public fun MediaAttachmentFactory(
+    maximumNumberOfPreviewedItems: Int = 4,
+    itemOverlayContent: @Composable (attachmentType: String?) -> Unit = { attachmentType ->
+        if (attachmentType == AttachmentType.VIDEO) {
+            DefaultItemOverlayContent()
+        }
+    },
+    previewItemOverlayContent: @Composable (attachmentType: String?) -> Unit = { attachmentType ->
+        if (attachmentType == AttachmentType.VIDEO) {
+            DefaultPreviewItemOverlayContent()
+        }
+    },
+): AttachmentFactory =
+    AttachmentFactory(
+        canHandle = {
+            it.none { attachment ->
+                !attachment.isImage() && !attachment.isVideo()
+            }
+        },
+        previewContent = { modifier, attachments, onAttachmentRemoved ->
+            MediaAttachmentPreviewContent(
+                attachments = attachments,
+                onAttachmentRemoved = onAttachmentRemoved,
+                modifier = modifier,
+                previewItemOverlayContent = previewItemOverlayContent
+            )
+        },
+        content = @Composable { modifier, state ->
+            MediaAttachmentContent(
+                modifier = modifier,
+                attachmentState = state,
+                maximumNumberOfPreviewedItems = maximumNumberOfPreviewedItems,
+                itemOverlayContent = itemOverlayContent
+            )
+        }
+    )
+```
+
+* `maximumNumberOfPreviewedItems`: Controls the maximum amount of tiles that can be displayed when previewing attachments inside the message list. In case this value is smaller than the number of attachments contained in the message, a count informing us how many more are left is displayed.
```

Comment:
```suggestion
* `maximumNumberOfPreviewedItems`: Controls the maximum amount of tiles that can be displayed when previewing attachments inside the message list. In case there are more attachments inside the message then we are previewing, a count of the remaining items is displayed.
```

Reply:
Used the given example and slightly built on top of it:

```
* `maximumNumberOfPreviewedItems`: Controls the maximum number of tiles that can be displayed when previewing attachments inside the message list. In case there are more attachments inside the message than are able to be previewed, the remaining item count is displayed.
```